pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -335,10 +335,25 @@ def plot_group(keys, values, ax):
             with plt.rc_context(rc):
                 ax = plt.gca()
         data = data._get_numeric_data()
-        if columns is None:
-            columns = data.columns
+
+        if columns:
+
+            # this is to align the current behavior of boxplot on columns, so if users
+            # explicitly specify column names and it is in df columns, then take the
+            # subset; Or if it is columns is in the first level, this is set mainly to
+            # avoid API change
+            if set(columns).issubset(data.columns) or set(columns).issubset(
```

Comment:
Hmm what’s the use case for just looking at the first level of the MultiIndex for label matching?